### PR TITLE
fix(gateway): Add in accidental missing generated types

### PR DESCRIPTION
**Pull request by trevor-scheer**
In review I missed the fact that we never added the new OOB file to the
`documents` list in our codegen config, which resulted in some missing
generated types. This fixes the configuration and adds the missing types
in the applicable locations.

### CODE REVIEW

**Review by Y-Guo — APPROVED** [review]

File: gateway-js/src/outOfBandReporter.ts [hunk 1]
```diff
@@ -11,6 +15,18 @@ export const OUT_OF_BAND_REPORTER_QUERY = /* GraphQL */`#graphql
 
 const { name, version } = require('../package.json');
 
+type OobReportMutationResult =
```
Comment (glasser) [inline gateway-js/src/outOfBandReporter.ts:18]: A little surprised that these types can't be easily created by graphql-code-generator, but looks like they can't be (yet). I guess those aren't the types returned via stuff like Apollo Client so it's not what is targeted.

**Review by glasser — APPROVED** [review]

File: gateway-js/src/outOfBandReporter.ts (hunk 1 above)
Comment (trevor-scheer) [reply gateway-js/src/outOfBandReporter.ts:18]: At the very least we could abstract this into a generic, not a bad idea. I think `GraphQLResponse` is a taken name but you get the idea.
`GraphQLResponse<OobReportMutationResult>`

File: gateway-js/src/outOfBandReporter.ts [hunk 2]
```diff
@@ -63,32 +79,34 @@ export class OutOfBandReporter {
...
-        message: error.message ?? error
-      },
-      request: {
-        url: request.url,
-        body: await request.text()
+    const variables: OobReportMutationVariables = {
+      input: {
+        error: {
+          code: errorCode,
+          message: error.message ?? error,
```
Comment (smyrick) [inline gateway-js/src/outOfBandReporter.ts:86]: @trevor-scheer is this a typo? I was attempting to update our version of `@apollo/gateway` from `0.24` to `0.31` and now I am getting a typescript error

```
$ tsc
node_modules/@apollo/gateway/src/outOfBandReporter.ts:86:11 - error TS2322: Type 'string | Error' is not assignable to type 'string'.
  Type 'Error' is not assignable to type 'string'.

86           message: error.message ?? error,
             ~~~~~~~

  node_modules/@apollo/gateway/src/__generated__/graphqlTypes.ts:28:3
    28   message?: Maybe<Scalars['String']>;
         ~~~~~~~
    The expected type comes from property 'message' which is declared here on type 'Error'
```

Comment (smyrick) [reply gateway-js/src/outOfBandReporter.ts:86]: It looks like the error actually first occurs in version `^0.30`

Comment (trevor-scheer) [reply gateway-js/src/outOfBandReporter.ts:86]: Thanks @smyrick, taking a look now. Agree the types are off, surprised we're not seeing the error on our end as well 🤔

Comment (trevor-scheer) [reply gateway-js/src/outOfBandReporter.ts:86]: Possibly a different TS version or config?

Ours recognizes `error.message` as a `string`, which means we'll (theoretically) never evaluate the `??`. cc @Y-Guo, I'm curious if the `?? error` is serving a runtime purpose and maybe our types are off further up (i.e. maybe `error` isn't an `Error` for sure?)

Comment (smyrick) [reply gateway-js/src/outOfBandReporter.ts:86]: We are running TS with `4.3.4`

```
  "compilerOptions": {
    "allowJs": true,
    "baseUrl": ".",
    "checkJs": true,
    "declaration": true,
    "declarationMap": true,
    "experimentalDecorators": true,
    "incremental": true,
    "lib": ["esnext"],
    "module": "commonjs",
    "noUnusedLocals": true,
    "esModuleInterop": true,
    "outDir": "dist",
    "sourceMap": true,
    "target": "es2020",
    "tsBuildInfoFile": ".tsbuildinfo"
  },
```

**Pull request merged**